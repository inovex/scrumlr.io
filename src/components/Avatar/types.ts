export const AVATAR_SKIN_COLORS = ["Tanned", "Yellow", "Pale", "Light", "Brown", "DarkBrown", "Black"] as const;
export type AvatarSkinColor = (typeof AVATAR_SKIN_COLORS)[number];

export const AVATAR_TOP_TYPES = [
  "Eyepatch",
  "Hat",
  "Hijab",
  "LongHairBigHair",
  "LongHairBob",
  "LongHairBun",
  "LongHairCurly",
  "LongHairCurvy",
  "LongHairDreads",
  "LongHairFrida",
  "LongHairFro",
  "LongHairFroBand",
  "LongHairMiaWallace",
  "LongHairNotTooLong",
  "LongHairShavedSides",
  "LongHairStraight",
  "LongHairStraight2",
  "LongHairStraightStrand",
  "NoHair",
  "ShortHairDreads01",
  "ShortHairDreads02",
  "ShortHairFrizzle",
  "ShortHairShaggyMullet",
  "ShortHairShortCurly",
  "ShortHairShortFlat",
  "ShortHairShortRound",
  "ShortHairShortWaved",
  "ShortHairSides",
  "ShortHairTheCaesar",
  "ShortHairTheCaesarSidePart",
  "Turban",
  "WinterHat1",
  "WinterHat2",
  "WinterHat3",
  "WinterHat4",
] as const;
export type AvatarTopType = (typeof AVATAR_TOP_TYPES)[number];

export const AVATAR_CLOTHE_COLORS = [
  "Black",
  "Blue01",
  "Blue02",
  "Blue03",
  "Gray01",
  "Gray02",
  "Heather",
  "PastelBlue",
  "PastelGreen",
  "PastelOrange",
  "PastelRed",
  "PastelYellow",
  "Pink",
  "Red",
  "White",
] as const;
export type AvatarClotheColor = (typeof AVATAR_CLOTHE_COLORS)[number];

export const AVATAR_GRAPHIC_TYPES = ["Bat", "Cumbia", "Deer", "Diamond", "Hola", "Pizza", "Resist", "Selena", "Bear", "SkullOutline", "Skull"] as const;
export type AvatarGraphicType = (typeof AVATAR_GRAPHIC_TYPES)[number];

export const AVATAR_CLOTHE_TYPES = ["BlazerShirt", "BlazerSweater", "CollarSweater", "GraphicShirt", "Hoodie", "Overall", "ShirtCrewNeck", "ShirtScoopNeck", "ShirtVNeck"] as const;
export type AvatarClotheType = (typeof AVATAR_CLOTHE_TYPES)[number];

export const AVATAR_HAIR_COLORS = ["Auburn", "Black", "Blonde", "BlondeGolden", "Brown", "BrownDark", "PastelPink", "Blue", "Platinum", "Red", "SilverGray"] as const;
export type AvatarHairColor = (typeof AVATAR_HAIR_COLORS)[number];

export const AVATAR_FACIAL_HAIR_COLORS = ["Auburn", "Black", "Blonde", "BlondeGolden", "Brown", "BrownDark", "Platinum", "Red"] as const;
export type AvatarFacialHairColor = (typeof AVATAR_FACIAL_HAIR_COLORS)[number];

export const AVATAR_FACIAL_HAIR_TYPES = ["Blank", "BeardMedium", "BeardLight", "BeardMajestic", "MoustacheFancy", "MoustacheMagnum"] as const;
export type AvatarFacialHairType = (typeof AVATAR_FACIAL_HAIR_TYPES)[number];

export const AVATAR_ACCESSORIES_TYPES = ["Blank", "Kurt", "Prescription01", "Prescription02", "Round", "Sunglasses", "Wayfarers"] as const;
export type AvatarAccessoriesType = (typeof AVATAR_ACCESSORIES_TYPES)[number];

export const AVATAR_EYE_TYPES = ["Close", "Cry", "Default", "Dizzy", "EyeRoll", "Happy", "Hearts", "Side", "Squint", "Surprised", "Wink", "WinkWacky"] as const;
export type AvatarEyeType = (typeof AVATAR_EYE_TYPES)[number];

export const AVATAR_EYEBROW_TYPES = [
  "Angry",
  "AngryNatural",
  "Default",
  "DefaultNatural",
  "FlatNatural",
  "FrownNatural",
  "RaisedExcited",
  "RaisedExcitedNatural",
  "SadConcerned",
  "SadConcernedNatural",
  "UnibrowNatural",
  "UpDown",
  "UpDownNatural",
] as const;
export type AvatarEyebrowType = (typeof AVATAR_EYEBROW_TYPES)[number];

export const AVATAR_MOUTH_TYPES = ["Concerned", "Default", "Disbelief", "Eating", "Grimace", "Sad", "ScreamOpen", "Serious", "Smile", "Tongue", "Twinkle", "Vomit"] as const;
export type AvatarMouthType = (typeof AVATAR_MOUTH_TYPES)[number];
