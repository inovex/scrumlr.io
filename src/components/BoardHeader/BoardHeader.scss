@import "constants/style";

.board-header {
  position: absolute;
  left: 30px;
  right: 30px;
  top: $column__border-width;
  height: $header__height;
  display: flex;
  flex-direction: row;
  align-items: center;
}

.board-header__link {
  position: absolute;
  left: 0;
  padding: $padding--small $padding--large;
  border-radius: 16px;
  width: max-content;
  cursor: pointer;
  transition: background-color 0.2s ease-in-out;
  outline: none;
  background-color: transparent;
  border: none;

  &:focus-visible {
    box-shadow: 0 0 0 2px rgba($color-backlog-blue, 0.32);
  }

  &:hover {
    background-color: rgba($color-backlog-blue, 0.06);
  }
}

[theme="dark"] {
  .board-header__link {
    &:focus-visible {
      box-shadow: 0 0 0 2px rgba($color-planning-pink, 0.48);
    }

    &:hover {
      background-color: rgba($color-planning-pink, 0.15);
    }
  }
}

.board-header__users {
  position: absolute;
  right: 0;
  padding: $padding--small;

  display: flex;
  flex-direction: row;
  align-items: center;

  background: none;
  border: none;
  outline: none;
}

.board-header_name-and-settings {
  position: absolute;
  left: 50%;
  transform: translateX(-50%);
  width: max-content;
  padding: $padding--small $padding--large;
  border-radius: 16px;
  transition: background-color 0.2s ease-in-out;

  outline: 0;
  border: 0;
  background: none;

  cursor: pointer;

  &:focus-visible {
    box-shadow: 0 0 0 2px rgba($color-backlog-blue, 0.32);
  }

  &:hover {
    background-color: rgba($color-backlog-blue, 0.06);
  }
}

[theme="dark"] {
  .board-header_name-and-settings {
    &:focus-visible {
      box-shadow: 0 0 0 2px rgba($color-planning-pink, 0.48);
    }

    &:hover {
      background-color: rgba($color-planning-pink, 0.15); // Ask designer for correct color
    }
  }
}

.board-header__access-policy-status {
  color: $color-middle-gray;
  font-size: $text-size--medium;
  font-weight: bold;
  letter-spacing: $letter-spacing--small;
  display: flex;
  align-items: center;
  width: max-content;
  margin: 0 auto;
}

.board-header__name-container {
  position: relative;
}

.board-header__name {
  display: inline-block;

  color: $color-black;
  font-size: 32px;
  font-weight: bold;
  letter-spacing: $letter-spacing--large;
  line-height: 40px;
  margin: 0;

  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  max-width: 28vw;
}

[theme="dark"] {
  .board-header__name {
    color: $color-white;
  }
}

.board-header__access-policy-status-icon {
  display: inline;
  vertical-align: bottom;

  width: $icon--medium;
  height: $icon--medium;
  align-self: center;
  margin-right: 2px;
}

.board-header__access-policy-status-icon * {
  color: $color-middle-gray;
}

@media #{$tablet} {
  .board-header {
    left: 60px;
    right: 60px;
  }

  .board-header__name-container {
    min-height: 40px;
    min-width: 240px;
  }
}

@media #{$smartphone} {
  .board-header__users {
    right: 0;
  }

  .board-header__link {
    padding: $padding--small;
  }
}
