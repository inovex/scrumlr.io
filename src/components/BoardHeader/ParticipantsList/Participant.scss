.participants__list-item .participant__user-avatar {
  margin: 10px 0 0 10px;
  height: 32px;
  width: 32px;
  font-size: 13px;
}
