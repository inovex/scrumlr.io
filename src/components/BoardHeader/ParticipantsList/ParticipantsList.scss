@import "constants/style.scss";

$header-bottom_distance: 16px;
$header-left_distance: 24px;

.participants {
  position: absolute;
  top: 92px;
  right: 80px;
  width: 300px;
  border-radius: 16px;
  box-shadow: 0 16px 32px 0 rgba(0, 87, 255, 0.24);
  overflow: hidden;
  z-index: 200;
}
@media #{$smartphone} {
  .participants {
    right: calc(50% - 150px);
  }
}

.participants__header {
  color: white;
  height: 86px;
  position: relative;
  background-color: $color-backlog-blue;
}

.participants__header-text {
  margin: 0;
  font-weight: bold;
  font-size: 18px;
  line-height: 18px;
  letter-spacing: 0.25px;
  position: absolute;
  top: 24px;
  left: $header-left_distance;
}

.participants__header-number {
  margin-top: 0;
  font-weight: bold;
  font-size: 12px;
  line-height: 18px;
  letter-spacing: 0.25px;
  position: absolute;
  top: -12px;
  right: -12px;
}

.participants__header-input {
  background-color: transparent;
  color: $color-white;
  font-size: 13px;
  letter-spacing: 0.27px;
  border: none;
  border-bottom: 1px solid transparent;
  position: absolute;
  bottom: $header-bottom_distance;
  left: $header-left_distance;
  padding-left: 24px;
  opacity: 0.8;

  &::placeholder {
    opacity: 1;
    color: white;
  }
}

.participants__header > input:focus {
  outline: none;
  border-bottom: 1px solid white;
}

.header__icon {
  position: absolute;
  bottom: $header-bottom_distance;
  left: $header-left_distance;
}

.participants__list {
  list-style-type: none;
  padding: 0;
  margin: 0;
  min-height: 80px;
  max-height: 272px;
  overflow: auto;
  scrollbar-color: $color-middle-gray #fff;
  background-color: #fff;
}

.participants__list::-webkit-scrollbar {
  width: 10px;
  background-color: #fff;
}

.participants__list::-webkit-scrollbar-track {
  border-radius: 8px;
}

.participants__list::-webkit-scrollbar-thumb {
  border-radius: 8px;
  background-color: $color-middle-gray;
}

.list__header {
  height: 32px;
  position: relative;
}

.list__header > label {
  letter-spacing: 0.18px;
  color: #a3a6aa;
  opacity: 0.8;
  font-size: 12px;
  position: absolute;
  top: 16px;
}

.list__header > label:nth-child(1) {
  text-align: left;
  left: 12px;
}
.list__header > label:nth-child(2) {
  text-align: right;
  left: 244px;
}

.participants__list-item {
  background-color: white;
  position: relative;
  height: 48px;
  padding-bottom: 52px;
}

.participants__list-item:not(:last-child)::after {
  content: "";
  height: 1px;
  width: calc(100% - 48px);
  background-color: rgba(35, 35, 35, 0.1);
  position: absolute;
  bottom: 0;
  left: 48px;
}

.participant__permission-toggle {
  transform: scale(0.9);
  left: 244px;
  top: 12px;
}

@media (prefers-color-scheme: dark) {
  .participants {
    box-shadow: 0 16px 32px 0 rgba(0, 0, 0, 0.2);
  }
  .participants__header {
    background-color: $color-planning-pink;
  }
  .participants__list {
    scrollbar-color: $color-darkest-gray $color-dark-mode-note;
    background-color: $color-dark-mode-note;
  }
  .participants__list::-webkit-scrollbar {
    background-color: $color-dark-mode-note;
  }
  .participants__list::-webkit-scrollbar-thumb {
    background-color: $color-darkest-gray;
  }
  .list__header {
    background: $color-dark-mode-note;
  }
  .participants__list-item {
    background-color: $color-dark-mode-note;
  }
  .participants__list-item:hover {
    background-color: #3d4555;
  }
  .participants__list-item:not(:last-child)::after {
    background-color: rgba(125, 125, 125, 0.1);
  }
}
