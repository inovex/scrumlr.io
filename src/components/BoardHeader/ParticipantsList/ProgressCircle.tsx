type ProgressCircleProps = {
  className?: string;
  percentage: number;
};

export const ProgressCircle = ({percentage, className}: ProgressCircleProps) => {
  // clamp between 0 and 1
  percentage = Math.min(Math.max(percentage, 0), 1);
  return (
    <svg viewBox="0 0 100 100" className={className}>
      <circle
        cx="50%"
        cy="50%"
        r="50%"
        stroke="currentColor"
        strokeWidth="100%"
        strokeDasharray={`${Math.PI * 100}%`}
        strokeDashoffset={`${Math.PI * 100 * (1 - percentage)}%`}
        transform="rotate (-90 50 50)"
      />
    </svg>
  );
};
