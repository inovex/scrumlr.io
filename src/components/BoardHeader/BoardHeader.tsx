import "./BoardHeader.scss";
import lock from "assets/icon-lock.svg";
import BoardUsers from "components/BoardUsers/BoardUsers";
import HeaderLogo from "./HeaderLogo/HeaderLogo";

export interface BoardHeaderProps {
  boardstatus: string;
  name: string;
}

const BoardHeader = ({name, boardstatus}: BoardHeaderProps) => (
  <div className="board-header">
    <HeaderLogo />
    <div className="board-header__infos">
      <p className="board-header__status">{boardstatus}</p>
      <div className="board-header__title-block">
        <img className="board-header__status-image" src={lock} alt="Private Session" />
        <h1 className="board-header__title">{name}</h1>
      </div>
    </div>
    <div className="board-header__users">
      <BoardUsers />
    </div>
  </div>
);
export default BoardHeader;
