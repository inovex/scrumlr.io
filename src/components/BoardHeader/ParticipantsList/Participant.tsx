import {UserClientModel} from "types/user";
import {UserAvatar} from "components/BoardUsers";
import {ToggleButton} from "components/ToggleButton";
import Parse from "parse";
import store from "store";
import {ActionFactory} from "store/action";
import {FunctionComponent} from "react";
import "./Participant.scss";

interface ParticipantProps {
  participant: UserClientModel;
  currentUserIsModerator: boolean;
  boardOwner: string;
}

export const Participant: FunctionComponent<ParticipantProps> = ({participant, currentUserIsModerator, boardOwner}) => (
  <li className="participants__list-item">
    <UserAvatar id={participant!.id} name={participant!.displayName} className="participant__user-avatar" />
    {currentUserIsModerator && (
      <ToggleButton
        className="participant__permission-toggle"
        disabled={Parse.User.current()?.id === participant!.id || participant!.id === boardOwner}
        values={["participant", "moderator"]}
        value={participant!.admin ? "moderator" : "participant"}
        onToggle={(val: "participant" | "moderator") => {
          store.dispatch(ActionFactory.changePermission(participant!.id, val === "moderator"));
        }}
      />
    )}
  </li>
);
