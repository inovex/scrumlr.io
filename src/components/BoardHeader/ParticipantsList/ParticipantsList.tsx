import Portal from "components/Portal/Portal";
import {useState} from "react";
import {UserClientModel} from "types/user";
import avatar from "assets/avatar.png";
import "./ParticipantsList.scss";
import {ToggleButton} from "components/ToggleButton";
import store, {useAppSelector} from "store";
import Parse from "parse";
import {ActionFactory} from "store/action";
import {ReactComponent as SearchIcon} from "assets/icon-search.svg";

type ParticipantsListProps = {
  open: boolean;
  onClose: () => void;
  participants: Array<UserClientModel>;
  currentUserIsModerator: boolean;
};

export const ParticipantsList = (props: ParticipantsListProps) => {
  const [searchString, setSearchString] = useState("");
  const boardOwner = useAppSelector((state) => state.board.data?.owner);

  if (!props.open) {
    return null;
  }

  return (
    <Portal onClose={props.onClose} darkBackground={false}>
      <aside className="participants">
        <header className="participants__header">
          <h4>Board Participants ({props.participants.length})</h4>
          <SearchIcon className="header__icon" />
          <input placeholder="Search" onChange={(event) => setSearchString(event.target.value.trim().toLowerCase())} />
        </header>
        <ul className="participants__list">
          <div className="list__header">
            <label>Name</label>
            {props.currentUserIsModerator && <label>Admin</label>}
          </div>
          {props.participants
            .sort((parA, parB) => parA.displayName.localeCompare(parB.displayName)) // Sort participants by name
            .filter((participant) => searchString.split(" ").every((substr) => participant.displayName.toLowerCase().includes(substr)))
            .map((participant) => (
              <li key={participant.id}>
                <figure>
                  <img src={avatar} />
                  <figcaption>{participant.displayName}</figcaption>
                </figure>
                {/* Show the permission toggle if the current user is moderator */}
                {props.currentUserIsModerator && (
                  <ToggleButton
                    className="participant__permission-toggle"
                    disabled={Parse.User.current()?.id === participant.id || participant.id === boardOwner}
                    values={["participant", "moderator"]}
                    value={participant.admin ? "moderator" : "participant"}
                    onToggle={(val: "participant" | "moderator") => {
                      store.dispatch(ActionFactory.changePermission(participant.id, val === "moderator"));
                    }}
                  />
                )}
              </li>
            ))}
        </ul>
      </aside>
    </Portal>
  );
};
