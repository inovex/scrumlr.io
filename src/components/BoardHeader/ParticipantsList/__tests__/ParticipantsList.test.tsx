import {render, fireEvent} from "@testing-library/react";
import {ActionFactory} from "store/action";
import {UserClientModel} from "types/user";
import {User} from "parse";
import {ParticipantsList} from "components/BoardHeader/ParticipantsList";
import {mocked} from "ts-jest/utils";
import * as store from "store";
import {useAppSelector} from "store";

jest.mock("store");
jest.mock("parse");

const mockedStore = mocked(store);
const mockedUseAppSelector = mocked(useAppSelector);
const mockedUser = mocked(User, true);

const createParticipantsList = (props: {open: boolean; onClose?: () => void; currentUserIsModerator: boolean; numberOfParticipants?: number}) => (
  <ParticipantsList
    open={props.open}
    onClose={() => props.onClose?.()}
    currentUserIsModerator={props.currentUserIsModerator}
    participants={[...Array(props.numberOfParticipants ?? 0).keys()].map((n) => ({id: `${n}`, displayName: `Participant ${n}`} as unknown as UserClientModel))}
  />
);

const createPortalAndParticipants = () => {
  const portal = global.document.createElement("div");
  portal.setAttribute("id", "portal");
  global.document.querySelector("body")!.appendChild(portal);
  return render(createParticipantsList({open: true, currentUserIsModerator: true, numberOfParticipants: 3}), {
    container: global.document.querySelector("#portal")!,
  });
};

describe("ParticipantsList", () => {
  beforeAll(() => {
    mockedUseAppSelector.mockResolvedValue({} as never);
  });

  describe("should render correctly", () => {
    test("on filtered participants", () => {
      const portal = global.document.createElement("div");
      portal.setAttribute("id", "portal");
      global.document.querySelector("body")!.appendChild(portal);
      const {container} = render(createParticipantsList({open: true, currentUserIsModerator: false, numberOfParticipants: 3}), {
        container: global.document.querySelector("#portal")!,
      });
      fireEvent.change(container.querySelector("input")!, {target: {value: "Participant 0"}});
      expect(container).toMatchSnapshot();
    });

    test("as moderator", () => {
      const portal = global.document.createElement("div");
      portal.setAttribute("id", "portal");
      global.document.querySelector("body")!.appendChild(portal);
      const {container} = render(createParticipantsList({open: true, currentUserIsModerator: true, numberOfParticipants: 3}), {
        container: global.document.querySelector("#portal")!,
      });
      expect(container).toMatchSnapshot();
    });

    test("on open", () => {
      const portal = global.document.createElement("div");
      portal.setAttribute("id", "portal");
      global.document.querySelector("body")!.appendChild(portal);
      const {container} = render(createParticipantsList({open: true, currentUserIsModerator: false, numberOfParticipants: 3}), {
        container: global.document.querySelector("#portal")!,
      });
      expect(container).toMatchSnapshot();
    });

    test("on close", () => {
      const {container} = render(createParticipantsList({open: false, currentUserIsModerator: false}), {container: global.document.querySelector("#portal")!});
      expect(container).toMatchSnapshot();
    });

    test("permission toggle should call store.dispatch", () => {
      mockedStore.default.dispatch = jest.fn();
      const portal = global.document.createElement("div");
      portal.setAttribute("id", "portal");
      global.document.querySelector("body")!.appendChild(portal);
      const {container} = render(createParticipantsList({open: true, currentUserIsModerator: true, numberOfParticipants: 3}), {
        container: global.document.querySelector("#portal")!,
      });
      fireEvent.click(container.getElementsByClassName("participant__permission-toggle")[0]);
      expect(mockedStore.default.dispatch).toHaveBeenCalled();
      expect(mockedStore.default.dispatch).toHaveBeenCalledWith(ActionFactory.changePermission("0", true));
    });

    test("should disable permission toggle of own user", () => {
      mockedUser.current = jest.fn(() => ({id: "0"} as never));
      const {container} = createPortalAndParticipants();
      expect(container.getElementsByClassName("participant__permission-toggle")[0]).toHaveAttribute("disabled");
    });

    test("should disable permission toggle of board creator", () => {
      mockedUseAppSelector.mockImplementation(() => "0" as never);
      const {container} = createPortalAndParticipants();
      expect(container.getElementsByClassName("participant__permission-toggle")[0]).toHaveAttribute("disabled");
    });
  });
});
