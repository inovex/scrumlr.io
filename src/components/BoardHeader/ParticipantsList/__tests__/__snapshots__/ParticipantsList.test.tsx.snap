// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipantsList should render correctly as moderator 1`] = `
<div
  id="portal"
>
  <div
    class="portal"
    role="dialog"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        class="portal__frame"
      >
        <div
          class="portal__content"
          role="dialog"
        >
          <aside
            class="participants"
          >
            <header
              class="participants__header"
            >
              <h4>
                Board Participants (
                3
                )
              </h4>
              <svg
                class="header__icon"
              >
                icon-search.svg
              </svg>
              <input
                placeholder="Search"
              />
            </header>
            <ul
              class="participants__list"
            >
              <div
                class="list__header"
              >
                <label>
                  Name
                </label>
                <label>
                  Admin
                </label>
              </div>
              <li>
                <figure>
                  <img
                    alt="Avatar of Participant 0"
                    src="avatar.png"
                  />
                  <figcaption>
                    Participant 0
                  </figcaption>
                </figure>
                <button
                  class="toggle-button toggle-button--left participant__permission-toggle"
                />
              </li>
              <li>
                <figure>
                  <img
                    alt="Avatar of Participant 1"
                    src="avatar.png"
                  />
                  <figcaption>
                    Participant 1
                  </figcaption>
                </figure>
                <button
                  class="toggle-button toggle-button--left participant__permission-toggle"
                />
              </li>
              <li>
                <figure>
                  <img
                    alt="Avatar of Participant 2"
                    src="avatar.png"
                  />
                  <figcaption>
                    Participant 2
                  </figcaption>
                </figure>
                <button
                  class="toggle-button toggle-button--left participant__permission-toggle"
                />
              </li>
            </ul>
          </aside>
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`ParticipantsList should render correctly on close 1`] = `<div />`;

exports[`ParticipantsList should render correctly on filtered participants 1`] = `
<div
  id="portal"
>
  <div
    class="portal"
    role="dialog"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        class="portal__frame"
      >
        <div
          class="portal__content"
          role="dialog"
        >
          <aside
            class="participants"
          >
            <header
              class="participants__header"
            >
              <h4>
                Board Participants (
                3
                )
              </h4>
              <svg
                class="header__icon"
              >
                icon-search.svg
              </svg>
              <input
                placeholder="Search"
              />
            </header>
            <ul
              class="participants__list"
            >
              <div
                class="list__header"
              >
                <label>
                  Name
                </label>
              </div>
              <li>
                <figure>
                  <img
                    alt="Avatar of Participant 0"
                    src="avatar.png"
                  />
                  <figcaption>
                    Participant 0
                  </figcaption>
                </figure>
              </li>
            </ul>
          </aside>
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`ParticipantsList should render correctly on open 1`] = `
<div
  id="portal"
>
  <div
    class="portal"
    role="dialog"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        class="portal__frame"
      >
        <div
          class="portal__content"
          role="dialog"
        >
          <aside
            class="participants"
          >
            <header
              class="participants__header"
            >
              <h4>
                Board Participants (
                3
                )
              </h4>
              <svg
                class="header__icon"
              >
                icon-search.svg
              </svg>
              <input
                placeholder="Search"
              />
            </header>
            <ul
              class="participants__list"
            >
              <div
                class="list__header"
              >
                <label>
                  Name
                </label>
              </div>
              <li>
                <figure>
                  <img
                    alt="Avatar of Participant 0"
                    src="avatar.png"
                  />
                  <figcaption>
                    Participant 0
                  </figcaption>
                </figure>
              </li>
              <li>
                <figure>
                  <img
                    alt="Avatar of Participant 1"
                    src="avatar.png"
                  />
                  <figcaption>
                    Participant 1
                  </figcaption>
                </figure>
              </li>
              <li>
                <figure>
                  <img
                    alt="Avatar of Participant 2"
                    src="avatar.png"
                  />
                  <figcaption>
                    Participant 2
                  </figcaption>
                </figure>
              </li>
            </ul>
          </aside>
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</div>
`;
